File: backend/app/schemas/workbench.py
from pydantic import BaseModel


class WorkbenchProjectItem(BaseModel):
    id: int
    project_no: str
    project_name: str
    client_name: str
    my_project_role: str | None = None
    project_leader_name: str | None = None
    transfer_user_name: str | None = None
    current_step: str
    status_display: str
    todo_action: str | None = None
    termination_status: str | None = None
    termination_reason: str | None = None
    delete_request_status: str | None = None
    delete_request_id: int | None = None
    delete_request_reason: str | None = None
    delete_approver_user_id: int | None = None
    delete_approver_user_name: str | None = None
    delete_requester_user_name: str | None = None
    delete_requested_at: str | None = None
    can_edit: bool
    can_delete: bool
    can_archive: bool
    can_request_termination: bool = False
    can_approve_termination: bool = False
    can_approve_delete: bool = False
    can_enter: bool
    is_reminded: bool = False
    remind_count_today: int = 0
    latest_remind_at: str | None = None


class WorkbenchResponse(BaseModel):
    my_projects: list[WorkbenchProjectItem]
    todo_projects: list[WorkbenchProjectItem]

File: frontend/src/api/workbench.ts
import http from './http'

export interface WorkbenchProjectItem {
  id: number
  project_no: string
  project_name: string
  client_name: string
  project_leader_name?: string | null
  transfer_user_name?: string | null
  current_step: string
  status_display: string
  todo_action?: string | null
  termination_status?: string | null
  termination_reason?: string | null
  delete_request_status?: string | null
  delete_request_id?: number | null
  delete_request_reason?: string | null
  delete_approver_user_id?: number | null
  delete_approver_user_name?: string | null
  delete_requester_user_name?: string | null
  delete_requested_at?: string | null
  can_edit: boolean
  can_delete: boolean
  can_archive: boolean
  can_request_termination: boolean
  can_approve_termination: boolean
  can_approve_delete?: boolean
  can_enter: boolean
  is_reminded?: boolean
  remind_count_today?: number
  latest_remind_at?: string | null
}

export async function getWorkbench() {
  const { data } = await http.get('/workbench')
  return data as { my_projects: WorkbenchProjectItem[]; todo_projects: WorkbenchProjectItem[] }
}
